cmd/kip/kubeconfig.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/elotl/kip/pkg/util/k8s"
+	"github.com/elotl/node-cli/manager"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+func maybeLoadKubeConfig(path string) (*clientcmdapi.Config, error) {
+	if path == "" {
+		return nil, fmt.Errorf("no kubeconfig provided; trying to continue")
+	}
+	kubeConfig, err := clientcmd.LoadFromFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("loading kubeconfig %s: %v; trying to continue", path, err)
+	}
+	return kubeConfig, nil
+}
+
+func loadOrCreateKubeConfig(path, mainKubeConfig, saSecret string, rm *manager.ResourceManager) (*clientcmdapi.Config, error) {
+	if path == "" && saSecret == "" {
+		return nil, fmt.Errorf("no kubeconfig or service account secret provided")
+	}
+	if path != "" {
+		if saSecret != "" {
+			return nil, fmt.Errorf("no kubeconfig or service account secret provided")
```

Comment:
Should this message be different since we are here in case of path not being null and service account secret not being null? Thanks.

Reply:
Correct, this was a copy/paste problem. Thanks!